app/services/work_history_with_breaks.rb

```diff
@@ -0,0 +1,80 @@
+class WorkHistoryWithBreaks
+  class BreakPlaceholder
+    def initialize(month_range:)
+      @month_range = month_range
+    end
+
+    def start_date
+      @month_range.first
+    end
+
+    def length
+      @month_range.count
+    end
+  end
+
+  def initialize(work_history)
+    @work_history = work_history.sort_by(&:start_date)
+    @work_history_with_breaks = []
+    @current_job = nil
+  end
+
+  def call
+    return [] if @work_history.empty?
+
+    @work_history.each { |job| @work_history_with_breaks << job_entry(job) }
+
+    months_in_timeline = month_range(
+      start_date: @work_history.first.start_date,
+      end_date: Time.zone.now - 1.month,
+    )
+    break_months_in_timeline = work_break_months(months_in_timeline)
+    breaks = work_breaks(break_months_in_timeline)
+    @work_history_with_breaks += breaks if breaks.any?
+
+    @work_history_with_breaks.sort_by! { |entry| entry[:entry].start_date }
+  end
+
+private
+
+  def job_entry(job)
+    { type: :job, entry: job }
+  end
+
+  def break_placeholder_entry(month_range)
+    { type: :break_placeholder, entry: BreakPlaceholder.new(month_range: month_range) }
+  end
+
+  def month_range(start_date:, end_date:)
+    (start_date.to_date..end_date.to_date).map(&:beginning_of_month).uniq
+  end
+
+  def work_break_months(months_in_timeline)
+    @work_history.each do |job|
+      job_end_date = job.end_date.nil? ? Time.zone.now : job.end_date
+      months_in_job_period = month_range(start_date: job.start_date, end_date: job_end_date)
+
+      months_in_timeline -= months_in_job_period
+    end
+
+    months_in_timeline
+  end
+
+  def work_breaks(break_months_in_timeline)
```

Comment:
There is some complexity here that makes it challenging to read this, but  I'm not sure how could you improve. I found challenging that some of these private methods have similar name e.g `work_break_months` and `work_breaks`.  So `work_break_months` returns a timeline, maybe you could call it `breaks_in_timeline`, but in the end, I'm not sure if it makes any difference

Reply:
I agree, will have a think about renaming!

Reply:
Did a bit of tinkering! Another :eyes:?